platform/server-bundle/src/main/resources/webapp/js/services.js

```diff
@@ -2,9 +2,274 @@
     'use strict';
 
     var uiServices = angular.module('uiServices', ['ngResource']);
-
     uiServices.factory('Menu', function($resource) {
         return $resource('module/menu');
     });
 
+    uiServices.service('BootstrapDialogManager', function () {
+        var modalsList = [];
+
+        this.open = function (dialog) {
+            modalsList.push(dialog);
+            if (modalsList.length > 1) {
+                modalsList[modalsList.length-2].close();
+            }
+            dialog.open();
+        };
+
+        this.close = function (dialog) {
+            dialog.close();
+            modalsList.splice(modalsList.length-1);
+            if (modalsList.length > 0) {
+                modalsList[modalsList.length-1].open();
+            }
+        };
+
+        this.remove = function () {
+            modalsList[modalsList.length-1].close();
+            modalsList.splice(modalsList.length-1);
+            if (modalsList.length > 0) {
+                modalsList[modalsList.length-1].open();
+            }
+        };
+    });
+
+    uiServices.service('LoadingModal', function () {
+        var dialog, open = false;
+
+        this.open = function () {
+            if (!open) {
+                dialog = new BootstrapDialog({
+                    message: function(dialogRef) {
+                        var $message = $('<div></div>'),
+                        pageToLoad = dialog.getData('pageToLoad');
+                        $message.load(pageToLoad);
+
+                        return $message;
+                    },
+                    data: {
+                        'pageToLoad': '../server/resources/partials/loading-splash.html'
+                    },
+                    closable: false,
+                    draggable: false
+                });
+
+                dialog.realize();
+                dialog.getModalHeader().hide();
+                dialog.getModalFooter().hide();
+                dialog.getModalContent().addClass('splash');
+                dialog.getModalContent().css('margin-top', '40%');
+                dialog.getModalBody().css('padding', '0');
+
+                dialog.open();
+                open = true;
+            }
+        };
+
+        this.close = function () {
+            if (open) {
+                dialog.close();
+            }
+            open = false;
+        };
+
+        this.isOpen = function () {
+            return open;
+        };
+    });
+
+    uiServices.factory('ModalFactory', function (LoadingModal, BootstrapDialogManager) {
```

Comment:
The ModalFactory shouldn't rely on the LoadingModal... lets keep them separate...